rllib/evaluation/sampler.py

```diff
@@ -263,12 +263,21 @@ def _env_runner(base_env, extra_batch_callback, policies, policy_mapping_fn,
     """
 
     try:
-        if not horizon:
+        # If Trainer's horizon is provided, force-set the env's
+        # `max_episode_steps` spec to that value (otherwise, Trainer's horizon
+        # setting will be violated).
+        print("base_env={}".format(base_env))
+        if horizon:
+            base_env.get_unwrapped()[0].spec.max_episode_steps = horizon
```

Comment:
It might be unexpected to force set horizon this way. How about we raise an error if a finite horizon is specified, and it happens to be greater than the internal time-limit. That would prevent unexpected clashes between environment and trainer configs.

Reply:
Yup, agree that it could cause confusion doing this under the radar. Ok, I'll produce an error instead.

Reply:
All fixed. Please take another look.